src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs

```diff
@@ -75,36 +76,43 @@ public class VsCredentialProviderImporter
             _initializer = initializer ?? Initialize;
         }
 
-        [Import("VisualStudioAccountProvider", typeof(IVsCredentialProvider), AllowDefault = true)]
-        public IVsCredentialProvider ImportedProvider { get; set; }
+        [ImportMany(typeof(IVsCredentialProvider))]
+        public IEnumerable<IVsCredentialProvider> ImportedProviders { get; set; }
 
         /// <summary>
         /// Plugin providers are entered loaded the same way as other nuget extensions,
         /// matching any extension named CredentialProvider.*.exe.
         /// </summary>
         /// <returns>An enumeration of plugin providers</returns>
-        public ICredentialProvider GetProvider()
+        public IReadOnlyCollection<ICredentialProvider> GetProviders()
         {
-            ICredentialProvider result = null;
+            var results = new List<ICredentialProvider>();
 
             try
             {
                 _initializer();
 
-                if (ImportedProvider != null)
+                if (ImportedProviders != null)
                 {
-                    result = new VsCredentialProviderAdapter(ImportedProvider);
+                    // Ensure imported providers ordering is deterministic
+                    foreach (var importedProvider in ImportedProviders.OrderBy(p => p.GetType().Name))
+                    {
+                        results.Add(new VsCredentialProviderAdapter(importedProvider));
```

Comment:
Can this ctor fail? If so, should we just skip the importedProvider and move to the next?